modules/extension/xsd/xsd-core/src/main/java/org/geotools/xsd/impl/jxpath/JXPathUtils.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2024, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.xsd.impl.jxpath;
+
+import java.util.Enumeration;
+import org.apache.commons.jxpath.FunctionLibrary;
+import org.apache.commons.jxpath.JXPathContext;
+import org.xml.sax.helpers.NamespaceSupport;
+
+/** Contains utility methods to create safe {@code JXPathContext} objects. */
+public class JXPathUtils {
+
+    /**
+     * Creates a {@code JXPathContext} that disables calling Java methods from XPath expressions.
+     *
+     * @param contextBean the root node object
+     * @param lenient whether the context is in lenient mode
+     * @return the context
+     */
+    public static JXPathContext newSafeContext(Object contextBean, boolean lenient) {
```

Comment:
Thanks for the clear javadocs I appreciate it.

Minor feedback: Consider `newDataContext` to accurately describe the restriction of no methods. In a more complicated security model such as GeoFence data traversal may be considered "unsafe" leading to this method name being misleading / incomplete.

Reply:
I really don't understand this comment since GeoFence has no dependence on gt-complex or commons-jxpath.

Reply:
I was trying to indicate that specific attributes may be restricted (using a geofence as an example of a plugin that offers this kind of functionality). 

So "safe" indicates no functions, but it would not be safe from using xpath to traverse data to access a  restricted columns.

As indicated this was minor feedback only.